### PR TITLE
Add node-problem-detector to cluster setup

### PR DESCRIPTION
**Pull request by linki**
Adds the node problem detector config files.

### CODE REVIEW

File: cluster/manifests/node-problem-detector/ds-node-problem-detector.yaml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        - "exec /node-problem-detector --logtostderr --system-log-monitors=/config/kernel-monitor.json,/config/kernel-monitor-filelog.json,/config/docker-monitor.json,/config/docker-monitor-filelog.json"
+        securityContext:
+          privileged: true
+        resources:
+          limits:
+            cpu: "200m"
+            memory: "100Mi"
+          requests:
+            cpu: "20m"
+            memory: "20Mi"
```
Comment (aermakov-zalando) [inline cluster/manifests/node-problem-detector/ds-node-problem-detector.yaml]: Overcommitting. :(

File: cluster/manifests/node-problem-detector/ds-node-problem-detector.yaml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    metadata:
+      labels:
+        application: node-problem-detector
+        version: v0.6.3
+    spec:
+      containers:
+      - name: node-problem-detector
+        image: k8s.gcr.io/node-problem-detector:v0.6.3
+        command:
+        - "/bin/sh"
```
Comment (aermakov-zalando) [inline cluster/manifests/node-problem-detector/ds-node-problem-detector.yaml]: This doesn't evaluate any variables, so going via `sh` isn't really needed.

File: cluster/manifests/node-problem-detector/ds-node-problem-detector.yaml [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  selector:
+    matchLabels:
+      application: node-problem-detector
+      version: v0.6.3
+  template:
+    metadata:
+      labels:
+        application: node-problem-detector
+        version: v0.6.3
+    spec:
```
Comment (aermakov-zalando) [inline cluster/manifests/node-problem-detector/ds-node-problem-detector.yaml:19]: Priority is missing, should probably be node-critical.

File: cluster/manifests/node-problem-detector/ds-node-problem-detector.yaml [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+            cpu: "20m"
+            memory: "20Mi"
+        env:
+        - name: NODE_NAME
+          valueFrom:
+            fieldRef:
+              fieldPath: spec.nodeName
+        volumeMounts:
+        - name: log
+          mountPath: /var/log
```
Comment (aermakov-zalando) [inline cluster/manifests/node-problem-detector/ds-node-problem-detector.yaml:55]: Shouldn't this be readonly?

Comment (linki) [reply cluster/manifests/node-problem-detector/ds-node-problem-detector.yaml:55]: The old config used to write logs to a file :open_mouth: in `/var/log` itself. Changed it.

File: cluster/manifests/node-problem-detector/ds-node-problem-detector.yaml (hunk 1 above)
Comment (linki) [reply cluster/manifests/node-problem-detector/ds-node-problem-detector.yaml]: fixed

File: cluster/manifests/node-problem-detector/ds-node-problem-detector.yaml (hunk 2 above)
Comment (linki) [reply cluster/manifests/node-problem-detector/ds-node-problem-detector.yaml]: everyone seems to run it like this for some reason. I changed to be more sane.

**Comment by linki** [comment]
:+1:

**Comment by muaazsaleem** [comment]
👍

**Pull request merged**